### PR TITLE
Visual Editor: ask the user to clear failed images on publish

### PR DESCRIPTION
**Pull request by maxme**
Fixes https://github.com/wordpress-mobile/WordPress-Editor-Android/issues/260

cc @aforcier

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -604,34 +606,22 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        // Show an Alert Dialog asking the user if he wants to remove all failed media before upload
         if (mEditorFragment.hasFailedMediaUploads()) {
-            ToastUtils.showToast(this, R.string.editor_toast_failed_uploads, Duration.SHORT);
-            return false;
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setMessage(R.string.editor_toast_failed_uploads)
+                    .setPositiveButton(R.string.editor_remove_failed_uploads, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            // Clear failed uploads
+                            mEditorFragment.removeAllFailedMediaUploads();
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: We could retry the save action here, but the user won't be able to verify his post looks good after the removal of failed media.

**Comment by aforcier** [comment]
Found an issue, not caused by your changes but we may as well fix it now:

If the editor is not in focus (such as opening an existing post, before the open PR fixing this is merged), and there are failed images, `removeAllFailedMediaUploads()` will clear the failed media but won't trigger the `media-removed` callback.

This is due to a null error: when the `MutationObserver` is tripped, [this call](https://github.com/wordpress-mobile/WordPress-Editor-Android/blob/dca82542b1c2521aece8cf8c428410b41100fc70/libs/editor-common/assets/ZSSRichTextEditor.js#L242) results in an error and the method doesn't complete.

We can fix this using some null checking in `ZSSEditor.getParentRangeOfFocusedNode`:

```javascript
ZSSEditor.getParentRangeOfFocusedNode = function() {
    var selection = window.getSelection();
    if (selection.focusNode == null) {
        return null;
    }
    return selection.getRangeAt(selection.focusNode.parentNode);
};
```

and checking for null before calling `ZSSEditor.setRange()` in `onMutationObserved` to avoid an error log.

**Comment by maxme** [comment]
Null checks added in 78382f2

**Comment by aforcier** [comment]
Looks great! :shipit:

**Pull request merged**